Tests/NIOTests/SocketAddressTest.swift

```diff
@@ -117,13 +117,14 @@ class SocketAddressTest: XCTestCase {
         }
     }
 
-    func testRejectsNonIPStrings() throws {
-        do {
-            _ = try SocketAddress(ipAddress: "definitelynotanip", port: 800)
-        } catch SocketAddressError.failedToParseIPString(let str) {
-            XCTAssertEqual(str, "definitelynotanip")
-        } catch {
-            XCTFail("Unexpected error \(error)")
+    func testRejectsNonIPStrings() {
+        XCTAssertThrowsError(try SocketAddress(ipAddress: "definitelynotanip", port: 800)) { error in
+            switch error as? SocketAddressError {
+            case .failedToParseIPString("definitelynotanip"):
```

Comment:
I think you need a `.some` for Swift 5.0 here

```suggestion
            case .some(.failedToParseIPString("definitelynotanip")):
```